apps/extension/src/final/extension-popup/components/trading-copilot-dialogs/trading-copilot-dialogs.tsx

```diff
@@ -0,0 +1,166 @@
+          )
+        })
+      }
+    </>
+  )
+};
+
+type TradingCopilotFormValues = {
+  amount: string;
+};
+
+const EMPTY_FORM: TradingCopilotFormValues = {
+  amount: '',
+};
+
+type TradingCopilotDialogProperties = {
+  dialog: {
+    details: {
+      name: string,
+      amount: number,
+      crypto: string,
+      when: string,
+      title: string,
+    }
+    user: {
+      balance: number,
+      crypto: string,
+    },
+    id: string,
+  },
+  activeDialogId: string | null,
+  closeDialog: () => void
+}
+
+const TradingCopilotDialog = ({dialog: {details, user, id}, activeDialogId, closeDialog}: TradingCopilotDialogProperties) => {
+  const form = useForm<TradingCopilotFormValues>({
+    defaultValues: EMPTY_FORM,
+  });
+
+  const avatarQuery = useCommandQuery({
+    command: new GetEnsInfoCommand({
+      ensName: details.name,
+      infoKey: 'avatar',
+    }),
+    staleTime: Number.POSITIVE_INFINITY,
+  });
+
+  return (
+    <Dialog open={id === activeDialogId} closeDialog={closeDialog} title={details.title}>
+      <div className="flex flex-col gap-y-6">
+        <div className='flex flex-row gap-3'>
+          <LazyImage
+            src={avatarQuery.data}
+            className="size-12 rounded-full"
+            fallbackComponent={
+              <Icon size={48} name="PersonIcon" className="rounded-full"/>
+            }
+          />
+          <div className="flex flex-col gap-2">
+            <p className="text-heading5 text-neutralGreen-900">{details.name} <span className="text-neutral-700 font-normal">purchased {details.amount} {details.crypto}</span></p>
+            <div className="flex flex-row gap-2 justify-between items-start">
+              <p className="text-label5 font-normal text-mint-700">{details.when}</p>
+            </div>
+          </div>
+        </div>
+        <form>
+          <div className="flex flex-row justify-between items-center">
+            <label
+              htmlFor="cryptoAmount"
+              className="block text-lg text-neutralGreen-900"
+            >
+              Amount
+            </label>
+            <p className="text-sm text-neutral-700">Balance: {user.balance} {user.crypto}</p>
+          </div>
+          <Controller
+            control={form.control}
+            name="amount"
+            render={({field}) => {
+              return (
+                <span className="mt-1 relative flex flex-row">
+                  <input
+                    {...field}
+                    type="number"
+                    id="cryptoAmount"
+                    className="[appearance:textfield] [&::-webkit-outer-spin-button]:appearance-none [&::-webkit-inner-spin-button]:appearance-none [&:focus+svg]:border-[#3b82f6] text-right block w-full rounded-md border border-[#D1D5DB] bg-white ps-14 pe-3 py-2 text-black shadow-sm focus:border-[#3b82f6] focus:outline-none focus:ring-[#3b82f6] sm:text-sm"
```

Comment:
nice way to handle the type number input 👌👌

Reply:
Actually, there is a numeric input built in our DS. Please check the `NumericInput`, we can use it and keep the arrows for native behavior.